src/current/v24.3/troubleshoot-lock-contention.md

```diff
@@ -0,0 +1,403 @@
+                                 | waiting                      | `UPDATE t SET v=2032 WHERE k=2;`
+                                 |                              | waiting
+`COMMIT;`                        | unblocked to proceed         | unblocked to proceed
+success, k=2,v=2012              | k=2, v=2012                  |
+                                 | `COMMIT;`                    | `COMMIT;`
+                                 | success                      | success                                 
+                                 |                              | `SELECT * FROM t WHERE k=2;`
+                                 |                              | k=2, v=2032
+
+### SQL statements
+
+To reproduce Example 1 in CockroachDB in preparation for the next section on how to [identify waiting and blocking transactions](#identify-waiting-and-blocking-transactions), execute the following SQL statements in the given order in the specified terminal.
+
+**Terminal 1**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BEGIN;
+UPDATE t SET v=2012 WHERE k=2; -- lock k=2
+~~~
+
+**Terminal 2**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BEGIN;
+SELECT * FROM t WHERE k=2; -- waiting read
+~~~
+
+**Terminal 3**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BEGIN;
+UPDATE t SET v=2032 WHERE k=2; -- waiting write
+~~~
+
+**Terminal 1**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+COMMIT;
+~~~
+
+**Terminal 2**
+
+When *Transaction 1* releases key `k=2`, *Transaction 2* should output the following:
+
+~~~
+  k |  v
+----+-------
+  2 | 2012
+(1 row)
+~~~
+
+`COMMIT` Transaction 2:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+COMMIT;
+~~~
+
+**Terminal 3**
+
+`COMMIT` Transaction 3 and verify that the `UPDATE` has succeeded:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+COMMIT;
+SELECT * FROM t where k=2;
+~~~
+
+The `SELECT` statement should output the following:
+
+~~~
+  k |  v
+----+-------
+  2 | 2032
+(1 row)
+~~~
+
+## Identify waiting and blocking transactions
+
+This section of the tutorial uses the [**Insights** page]({% link {{ page.version.version }}/ui-insights-page.md %}#transaction-executions-view) of the DB Console to identify waiting and blocked transactions in the demo cluster. With a CockroachDB {{ site.data.products.cloud }} cluster, the {{ site.data.products.cloud }} Console has a similar [**Insights** page]({% link cockroachcloud/insights-page.md %}#transaction-executions-view). You can also use the [`crdb_internal`]({% link {{ page.version.version }}/performance-recipes.md %}#identify-transactions-and-objects-that-experienced-lock-contention) system catalog to view tables and indexes that experienced contention.
+
+This section of the tutorial assumes you have already run the SQL statements from [Example 1](#example-1). When troubleshooting lock contention in your own workload, you can adapt the following steps using the DB Console or the {{ site.data.products.cloud }} Console.
```

Comment:
Really small nit, it would be great to rephrase the start of one of these paragraphs to avoid the double "This section..." but not a huge deal.

Reply:
updated